Assets/_BForBoss/_Weapons/Scripts/Weapons/Weapon.cs

```diff
@@ -111,9 +113,9 @@ private void Fire()
              OnFireWeapon?.Invoke(_weaponProperties.BulletsPerShot);
          }
 
-         private Quaternion GenerateSpreadAngle()
+         private Quaternion GenerateSpreadAngle(float timeSinceFiring)
          {
-             var spread = _weaponProperties.BulletSpread;
+             var spread = _weaponProperties.BulletSpread * _weaponProperties.GetBulletSpreadWeight(timeSinceFiring);
```

Comment:
`_weaponProperties.GetBulletSpreadWeight(timeSinceFiring);` separate method for this

Reply:
Changes Made